src/v/coproc/tests/coproc_test_fixture.cc

```diff
@@ -98,36 +78,64 @@ ss::future<coproc_test_fixture::opt_reader_data_t> coproc_test_fixture::drain(
     });
 }
 
+void interpret_batch(
+  std::size_t& n_records_read,
+  model::offset& offset,
+  model::record_batch_reader::data_t& data,
+  model::record_batch_reader::data_t&& batch) {
+    if (batch.empty()) {
+        return;
+    }
+    offset = batch.back().last_offset();
+    for (model::record_batch& r : batch) {
+        n_records_read += r.record_count();
+        data.push_back(std::move(r));
+    }
+}
+
+model::offset next_offset(model::offset o) {
```

Comment:
Why not initialise the `state::offset` above as `{-1}` and drop this function?

Reply:
Oh, looks like that might not work, if `pw.last_stable_offset()` is initialized as `::min()`

Reply:
Yeah i can actually just start at 0